tests/solve_group_tests.rs

```diff
@@ -185,3 +186,167 @@ async fn test_compressed_mapping_catch_missing_package() {
+        &conda_mapping,
+        &compressed_mapping,
+    )
+    .unwrap();
+
+    pypi_mapping::prefix_pypi_name_mapping::amend_pypi_purls_for_record(
+        &mut boltons_repo_data_record,
+        &conda_mapping,
+        &compressed_mapping,
+    )
+    .unwrap();
+
+    // package is not yet present
+    // so just assume that conda name == pypi name
+    let first_purl = repo_data_record.package_record.purls.pop().unwrap();
+
+    assert!(first_purl.name() == "pixi-something-new-for-test");
+
+    let boltons_purl = boltons_repo_data_record.package_record.purls.pop().unwrap();
+
+    assert!(boltons_purl.qualifiers().get("source").unwrap() == "conda-forge-mapping");
+}
+
+#[tokio::test]
+async fn test_we_record_not_present_package_as_purl_for_custom_mapping() {
+    let pixi = PixiControl::from_manifest(&format!(
+        r#"
+    [project]
+    name = "test-channel-change"
+    channels = ["conda-forge"]
+    platforms = ["linux-64"]
+    conda-pypi-map = {{ 'conda-forge' = "https://raw.githubusercontent.com/prefix-dev/parselmouth/main/files/compressed_mapping.json" }}
+
+    "#,
+    ))
+    .unwrap();
+
+    let project = pixi.project().unwrap();
+
+    let client = project.authenticated_client();
+    let foo_bar_package = Package::build("pixi-something-new", "2").finish();
+    let boltons_package = Package::build("boltons", "2").finish();
+
+    let repo_data_record = RepoDataRecord {
+        package_record: foo_bar_package.package_record,
+        file_name: "pixi-something-new".to_owned(),
+        url: Url::parse("https://pypi.org/simple/pixi-something-new-new/").unwrap(),
+        channel: "https://conda.anaconda.org/conda-forge/".to_owned(),
+    };
+
+    let boltons_repo_data_record = RepoDataRecord {
+        package_record: boltons_package.package_record,
+        file_name: "boltons".to_owned(),
+        url: Url::parse("https://pypi.org/simple/boltons/").unwrap(),
+        channel: "https://conda.anaconda.org/conda-forge/".to_owned(),
+    };
+
+    let mut packages = vec![repo_data_record, boltons_repo_data_record];
+
+    let mapping_map = project.pypi_name_mapping_source().custom().unwrap();
+
+    println!("mapping map is {:?}", mapping_map);
+
+    pypi_mapping::custom_pypi_mapping::amend_pypi_purls(&client, &mapping_map, &mut packages, None)
+        .await
+        .unwrap();
+
+    // println!("Purls are {:?}", repo_data_record.package_record.purls);
+
+    // package is not yet present
+    // so just assume that is the same as pypi
+
+    let mut boltons_package = packages.pop().unwrap();
+
+    let boltons_first_purl = boltons_package.package_record.purls.pop().unwrap();
+
+    assert!(boltons_first_purl.name() == "boltons");
+    assert!(boltons_first_purl.qualifiers().get("source").unwrap() == "project-defined-mapping");
+
+    let mut package = packages.pop().unwrap();
+
+    let first_purl = package.package_record.purls.pop().unwrap();
+
+    assert!(first_purl.name() == "pixi-something-new")
```

Comment:
Again could use some more comments at to what this is testing. It's hard to get into if you look at it with fresh eyes :)